terraform/ec2/main.tf

```diff
@@ -124,7 +127,7 @@ data "template_file" "mocked_server_cert_for_windows" {
 }
 resource "null_resource" "setup_mocked_server_cert_for_windows" {
   depends_on = [null_resource.check_patch]
-  count = local.selected_ami["family"] == "windows" ? 1 : 0
+  count = local.selected_ami["family"] == "windows" && !var.ssm_mode ? 1 : 0
```

Comment:
Why we want to disable setup mocked server cert when using ssm?

Reply:
Actually SSM testcase don't use mocked server instance. It will only check whether ADOT Collector is running in aoc instance.

Reply:
I think it's better to add a var to disable mocked server to common e.g. `var.disable_mocked_server`, because there are other tests that don't require mocked server. Then when testing ssm you can set that variable to true. I am not sure if there is one hanging around, though since ssm is only for ec2 I think using ssm_mode is fine as well.

Reply:
Good idea. I add `disable_mocked_server` and `enable_ssm_validate` to replace `ssm_mode`. All new variable definitions are moved into terraform/ec2/variables.tf because it is only for ec2 test.